core/src/main/java/org/infinispan/configuration/ConfigurationManager.java

```diff
@@ -104,12 +113,23 @@ public void removeConfiguration(String cacheName) {
    public Collection<String> getDefinedCaches() {
       List<String> cacheNames = namedConfiguration.entrySet().stream()
             .filter(entry -> !entry.getValue().isTemplate())
-            .map(entry -> entry.getKey())
+            .map(Map.Entry::getKey)
             .collect(Collectors.toList());
       return Collections.unmodifiableCollection(cacheNames);
    }
 
    public Collection<String> getDefinedConfigurations() {
       return Collections.unmodifiableCollection(namedConfiguration.keySet());
    }
+
+   private boolean isInvalid(Map.Entry<String, Configuration> entry) {
+      try {
+         gcr.notifyCreatingConfiguration(entry.getValue());
+         return false;
+      } catch (Throwable t) {
+         //it throws an exception if it isn't valid
+         CONFIG.removedInvalidConfiguration(entry.getKey(), t);
```

Comment:
This seems the wrong place to log that a configuration will be removed, the method name suggests that it is idempotent.

Reply:
The method is only used by `removeIf`. I can change the method name if you want.